### PR TITLE
Add a scraper for german indeed

### PR DESCRIPTION
**Pull request by marchbnr**
# Add a scraper for german indeed

## Description

Implemented a scraper implementation for the german indeed website.
The functionality includes a subset of the changes suggested in #132 by @Luckyz7 and the required changes were commented. A different locale name was used to align more closely with iso codes.

## Context of change

Please add options that are relevant and mark any boxes that apply.

- [ ] Software (software that runs on the PC)
- [x] Library (library that runs on the PC)
- [ ] Tool (tool that assists coding development)
- [ ] Other

## Type of change

Please mark any boxes that apply.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## How Has This Been Tested?

Tested manually using a configuration in the demo directory.

## Checklist:

Please mark any boxes that have been completed.

- [x] I have performed a self-review of my own code.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] I have made corresponding changes to the documentation.
- [x] My changes generate no new warnings.
- [ ] I have added tests that prove my fix is effective or that my feature works.
- [x] New and existing unit tests pass locally with my changes.
- [x] Any dependent changes have been merged and published in downstream modules.

### CODE REVIEW

**Comment by thebigG** [comment]
This is awesome, thanks so much for the contribution! But do you mind adding a test for your new demo file to https://github.com/PaulMcInnis/JobFunnel/blob/master/.github/workflows/ci.yml?

When you add it,  github Actions will automatically test your Germany scraper every time there is a push :).

**Comment by marchbnr** [comment]
> This is awesome, thanks so much for the contribution! But do you mind adding a test for your new demo file to https://github.com/PaulMcInnis/JobFunnel/blob/master/.github/workflows/ci.yml?
> 
> When you add it, github Actions will automatically test your Germany scraper every time there is a push :).

Thanks for the feedback. I have added a test run in the ci build, but the pipeline already fails due to existing issues.

**Comment by thebigG** [comment]
Thanks for adding it to the CI. Yes, we have issues in the CI. In fact me and @PaulMcInnis have been talking about this on #133. I thought I had fixed the issue with #134, but sadly it looks like it wasn't fixed completely :(. I'll try to look into it when I get time.

**Comment by thebigG** [comment]
I think at least now we won't have job_id issues anymore. What's left is deciding what to do about the error code when we don't find any jobs.

**Comment by marchbnr** [comment]
Cool, thank you for your changes! I think the status code should not be dependent on the number of results. One possibility to do this differently is by tracking for errors along the way and setting the status code after execution accordingly. However this is not related to the current feature, so I would do this in another pull request, if you agree.

File: demo/settings_DE.yaml [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+  locale: GERMANY_GERMAN
+
+  # Job providers which we will search, one of INDEED, MONSTER, GLASSDOOR:
+  # NOTE: we choose domain via locale (i.e. CANADA_ENGLISH -> www.indeed.ca)
+  # FIXME: we need to add back GLASSDOOR when that's working again.
+  providers:
+    - INDEED
+
+  # Region that we are searching for jobs within:
+  province_or_state: "108"  # NOTE: this is generally 2 characters long.
```
Comment (PaulMcInnis) [inline demo/settings_DE.yaml:27]: can you elaborate on what this string is? I think the comment is wrong for this now, what does 108 refer to?

File: demo/settings_DE.yaml [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+  city: "Berlin"  # NOTE: this is the full city / town name.
+  radius: 0  # km (NOTE: if we were in locale: USA_ENGLISH it's in miles)
+
+  # These are the terms you would be typing into the website's search field:
+  keywords:
+    - community-manager
```
Comment (PaulMcInnis) [inline demo/settings_DE.yaml:33]: Maybe we can change this to match the other pre-configured configs by searching for just Python? This helps with CI as we are almost always guaranteed to have results.

**Review by PaulMcInnis — APPROVED** [review]
LGTM! Just some minor comments, but I think the improved keying will help us avoid issues in the future.

We should also rev up the major version as we are introducing a breaking change for users who wish to continue an existing search by changing the keying.

**Comment by PaulMcInnis** [comment]
also once this goes in we should cut a new release as I think some of our recent issues are resolved by the current master.

File: demo/settings_DE.yaml (hunk 1 above)
Comment (marchbnr) [reply demo/settings_DE.yaml]: You are right. I had mistaken this to be an internal indeed code. This variable actually does not make sense for the german site and it is also not used to build the search url. I updated the config and comment to highlight, that it is not used and loosened the config validation to allow for empty strings. The same also applies to the UK indeed scraper.

File: demo/settings_DE.yaml (hunk 2 above)
Comment (marchbnr) [reply demo/settings_DE.yaml]: I changed this to the same keywords of the US indeed scraper. There are around 100 results and around 3 pages, thereby testing crawling of pagination while not crawling for a longer period.

**Comment by PaulMcInnis** [comment]
OK, going to merge this and cut a release with removed brotli encoding for all the other scrapers as well since it seems to be causing issues all around. Was a bit pre-emptive with 3.0.2

**Pull request merged**